dotnet/src/Connectors/Connectors.Memory.Redis/RedisMemoryStore.cs

```diff
@@ -27,14 +27,46 @@ public sealed class RedisMemoryStore : IMemoryStore
 {
     /// <summary>
     /// Create a new instance of semantic memory using Redis.
+    /// <see href="https://redis.io/docs/interact/search-and-query/search/vectors/#creation-attributes-per-algorithm"/>
+    /// <see href="https://redis.io/docs/interact/search-and-query/search/vectors/#querying-vector-fields"/>
     /// </summary>
-    /// <param name="database">The database of the redis server.</param>
-    /// <param name="vectorSize">Embedding vector size</param>
-    public RedisMemoryStore(IDatabase database, int vectorSize)
+    /// <param name="database">The database of the Redis server.</param>
+    /// <param name="vectorSize">Embedding vector size, defaults to 1536</param>
+    /// <param name="vectorIndexAlgorithm">Indexing algorithm for vectors, defaults to "HNSW"</param>
+    /// <param name="vectorType">Vector type, defaults to "FLOAT32"</param>
+    /// <param name="vectorDistanceMetric">Metric for measuring vector distances, defaults to "COSINE"</param>
+    /// <param name="queryDialect">Query dialect, must be 2 or greater for vector similarity searching, defaults to 2</param>
+    public RedisMemoryStore(IDatabase database, int vectorSize = 1536, string vectorIndexAlgorithm = "HNSW", string vectorType = "FLOAT32", string vectorDistanceMetric = "COSINE", int queryDialect = 2)
     {
         this._database = database;
         this._vectorSize = vectorSize;
         this._ft = database.FT();
+        this._vectorIndexAlgorithm = vectorIndexAlgorithm == "HNSW" ? Schema.VectorField.VectorAlgo.HNSW : Schema.VectorField.VectorAlgo.FLAT;
+        this._vectorType = vectorType;
+        this._vectorDistanceMetric = vectorDistanceMetric;
+        this._queryDialect = queryDialect;
+    }
+
+    /// <summary>
+    /// Create a new instance of semantic memory using Redis.
+    /// <see href="https://redis.io/docs/interact/search-and-query/search/vectors/#creation-attributes-per-algorithm"/>
+    /// <see href="https://redis.io/docs/interact/search-and-query/search/vectors/#querying-vector-fields"/>
+    /// </summary>
+    /// <param name="connectionString">Provide connection URL to a Redis instance</param>
+    /// <param name="vectorSize">Embedding vector size, defaults to 1536</param>
+    /// <param name="vectorIndexAlgorithm">Indexing algorithm for vectors, defaults to "HNSW"</param>
+    /// <param name="vectorType">Vector type, defaults to "FLOAT32"</param>
+    /// <param name="vectorDistanceMetric">Metric for measuring vector distances, defaults to "COSINE"</param>
```

Comment:
I'm wondering how we can minimize error probability for these parameters. Since they are strings, user can pass any value. 
1. We can list all possible values in XML documentation, so user will know which values can be accepted.
2. We can convert their type from `string` to `Enum`, in this case it won't be possible to pass incorrect value. 

I would probably go with `Enum` option in C#. It has disadvantages as well, for example as soon as Redis will accept new value, we will have to add new enum value. But looking at nature of these parameters, I don't think they will be updated a lot. While this will make our class more stable.

Reply:
Good idea, we can use `Enum.TryParse` with the `Schema.VectorField.VectorAlgo`  for interoperability with Redis, since they define VectorAlgo with an enum as well.
I'll add enums to a new file called `RedisVectorAttributes.cs` so its easier to find valid values.